types/underscore/underscore-tests.ts

```diff
@@ -152,187 +152,205 @@ declare const maybeFunction: (() => void) | undefined;
 
-var odds = _.reject([1, 2, 3, 4, 5, 6], (num) => num % 2 == 0);
-
-_.every([true, 1, null, 'yes'], x => !!_.identity(x));
+// determining whether every value is truthy
+_.every([true, 1, null, 'yes']);
 
 _.any([null, 0, 'yes', false]);
 
+// determining whether any number in a list is divisible by three
 _.some([1, 2, 3, 4], l => l % 3 === 0);
 
+// determining whether any value in a dictionary is uppercase
 _.some({ a: 'a', b: 'B', c: 'C', d: 'd' }, l => l === l.toUpperCase());
 
+// checking whether an item is in an array
 _.contains([1, 2, 3], 3);
 
+// checking whether an item is in the portion of an array that starts with index 1
 _.contains([1, 2, 3], 3, 1);
 
-_.invoke([[5, 1, 7], [3, 2, 1]], 'sort');
-
-// https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33479
-var foo: any[] = [{'a': 1, 'b': 2}];
-_.pluck(foo, 'a');
+// truncating a set of strings to 5 characters or less
+_.invoke(['zebra', 'giraffe', 'lion'], 'substring', 0, 5);
 
+// retrieving a property value from all items in a collection
 var stooges = [{ name: 'moe', age: 40 }, { name: 'larry', age: 50 }, { name: 'curly', age: 60 }];
 _.pluck(stooges, 'name');
 
-_.max(stooges, (stooge) => stooge.age);
-_.min(stooges, (stooge) => stooge.age);
-_.max({ a: 1, b: 2 });
-_.max({ a: 'a', b: 'bb' }, (v, k) => v.length);
+// retrieving the minimum number in a dictionary
 _.min({ a: 1, b: 2 });
-_.min({ a: 'a', b: 'bb' }, (v, k) => v.length);
 
-var numbers = [10, 5, 100, 2, 1000];
-_.max(numbers);
-_.min(numbers);
+// retrieving the item with the maximum number in a property
+_.max(stooges, (stooge) => stooge.age);
 
+// sorting by a calculated value
 _.sortBy([1, 2, 3, 4, 5, 6], (num) => Math.sin(num));
 
-_([1, 2, 3]).chain()
-    .sortBy(x => -x)
-    .sortBy(x => -x)
-    .value().length;
-
+// grouping numbers by their non-fractional parts
 _([1.3, 2.1, 2.4]).groupBy((e) => Math.floor(e));
-_.groupBy([1.3, 2.1, 2.4], (num) => Math.floor(num).toString());
-_.groupBy(['one', 'two', 'three'], 'length');
-
-_.indexBy(stooges, 'age')['40'].age;
-_(stooges).indexBy('age')['40'].name;
-_(stooges)
-    .chain()
-    .indexBy('age')
-    .value()['40'].age;
 
-let pensioners: string[] = _.chain(stooges)
-    .filter(p => p.age >= 60)
-    .map(p => p.name)
-    .value();
-
-var usersData: _.Dictionary<{ age: number; name: string }> = {
-    'user id': { name: 'moe', age: 40 },
-    'other user Id': { name: 'larry', age: 50 },
-    'fake id': { name: 'curly', age: 60 },
-};
-
-let youngPeopleId: string[] = _.chain(usersData)
-    .map((p, k: string) => k)
-    .value();
-
-let usersTable: { age: number; name: string; id: string }[] = _.chain(usersData)
-    .map((p, k: string) => {
-        return { id: k, ...p };
-    })
-    .value();
-
-// Test map function with _ChainOfArrays<>
-let usersTable_2 /*: { age: number; name: string; id: string }[][]*/ = _.chain(usersData)
-    .map((p, k: string) => {
-        return [{ id: k, ...p }];
-    })
-    .value();
```

Comment:
Removing lots of chain tests here. Are these all redundant?

Reply:
I see a lot of chain tests at the end of the diff, so I guess these were either redundant or moved.

Reply:
I considered any chain test with only one call in the chain to be redundant with combinatorial tests. Double-checking the ones I removed, I believe all of them except one for `values` at https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfL563 are covered as described. https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfR623 does at least provide a chain test that involves `values`, but I could put this back to make up for the current absence of a combinatorial single-call test for `values` if you'd like.

Reply:
With regards to chain tests that include more than one call, while the exact order of calls for the removed tests may not be represented in the final test set I've made sure that each function involved in them is included in one or more of the set of pre-existing and added multi-call chain tests. A fair number of the removed tests were also near-duplicates from a call chain standpoint; see kept test https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfR484-R490 vs. the removed tests below:
https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfL234-L237
https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfL443-L447
https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfL513-L516
https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfL518-L523
https://github.com/reubenrybnik/DefinitelyTyped/pull/1/files#diff-4d35bcc89b6520b7b1930f73c7e9c4bfL525-L530